Color contrast improvements for accessibility

ggeisler opened the pull request:
A couple of minor color contrast updates for things that didn't have sufficient contrast for accessibility purposes:

Links don't have sufficient contrast against the repository box background color. No lighter gray, in fact, provides sufficient contrast. So I went with removing the background color. We can assume implementers will style these differently to match their local style, anyway.

### Before
<img width="1165" alt="Screen Shot 2023-01-17 at 5 26 11 PM" src="https://user-images.githubusercontent.com/101482/213047927-c27d21c8-cd47-45ce-bc5e-acfce649e065.png">

### After
<img width="1175" alt="Screen Shot 2023-01-17 at 4 19 06 PM" src="https://user-images.githubusercontent.com/101482/213048003-109fc2f9-1954-45ff-a487-b353f4733275.png">

---

Same thing with the online content box. No lighter version of green provides sufficient contrast with the default Bootstrap blue link. Basically same solution as above.

### Before
<img width="770" alt="Screen Shot 2023-01-17 at 5 25 16 PM" src="https://user-images.githubusercontent.com/101482/213048585-aedb2a90-ca38-4b10-a0fa-749a22ab0a59.png">


### After
<img width="770" alt="Screen Shot 2023-01-17 at 5 24 59 PM" src="https://user-images.githubusercontent.com/101482/213048648-765de00c-9033-4195-8f4e-d68e7b5774dd.png">

jcoyne commented on app/assets/stylesheets/arclight/modules/hierarchy_and_online_contents.scss:
```diff
@@ -141,6 +141,10 @@ $hierarchy-view-collapse-icon: url("data:image/svg+xml,<svg xmlns='http://www.w3
   .online-contents {
     grid-area: online-contents;
+
+    .alert {
```
This to me seems like we're using alert improperly.  These are cards, not alerts.  Can we remove the class from the template instead?

jcoyne replied:
Can we style `.alert-dark` instead here?

ggeisler commented:
I've updated the online content box to use `card` rather than `alert`. 

(I also added a hidden span for the box title because I noticed in some other accessibility testing that we have duplicate headings with the text "Online content". The hidden text will help screenreader users by adding "Filter" to the first heading's text and remove the duplication.)

It looks pretty much like it did previously:

<img width="765" alt="Screen Shot 2023-01-18 at 1 35 51 PM" src="https://user-images.githubusercontent.com/101482/213290226-c113de7e-c065-4de6-9427-d9b153340a9c.png">

jcoyne reviewed (approved):
Thanks. This is much clearer.
